applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -820,34 +850,35 @@ class TrilinosBlockBuilderAndSolver
         // defining a temporary vector to gather all of the values needed
         Epetra_IntVector fixed(rA.ColMap());
 
-        // importing in the new temp vector the values
+        // Detect if there is a line of all zeros and set the diagonal to a 1 if this happens
+        const auto& r_process_info = rModelPart.GetProcessInfo();
+        const auto& r_data_comm = rModelPart.GetCommunicator().GetDataCommunicator();
+        mScaleFactor = TSparseSpace::CheckAndCorrectZeroDiagonalValues(r_process_info, rA, rb, r_data_comm, mScalingDiagonal);
```

Comment:
this does a lot of computation even if no scaling is applied
can you please refactor?

Reply:
What do you mean?, if no scaling it puts 1.0 on the diagonal of empty rows, that's necesssary.

Reply:
my issue is that even if I use `no_scaling` this function modifies the matrix, i.e. this is a behavior change!

Reply:
Yes, it is doing the same we do in serial, that's why the PR is called "Unify behavior between the serial/MPI BlockBuilderandSolver", and it specifies that in the description. @pooyan-dadvand agrees with this unification

Reply:
ok, I removed my block